### ISSUE TITLE
dump on a MongoDBConnect object will fetch the whole DB

### ISSUE THREAD

**Issue by MarkusWollny** [lucee/extension-mongodb#37]
In an attempt to avoid connection leaks and optimize connection reuse, I stored the result of MongoDBConnect inside the APPLICATION scope using a singleton pattern. This worked fine (in development, haven't tested in production yet), until a colleague attempted to cfdump the APPLICATION scope and crashed his Tomcat instance every time my component was instanced between the last Tomcat restart and the dump attempt.

Investigating the cfdump issue with a smaller database, I discovered that a dump of the supposed handle actually tries to fetch and dump the associated Mongo database. All of it. Which usually is not a good idea at all. So I wonder if this behaviour is actually working as intended or if the Mongo extension should in fact hook some reflection magic into the cfdump attempt to serialize the object for output and maybe dump methods and properties of the handle, so it will not in fact fetch all of the database.

So in fact when I do this:
```
dump(
	MongoDBConnect(
		"mongorocks",
		"localhost",
		27017
	)
);
```
I expected a dump of the available methods and properties, not the actual data inside the mongorocks database:
![image](https://user-images.githubusercontent.com/63056079/201107519-b13e1399-2d42-4eaa-ae72-22d5394b06ef.png)

**Comment by sjdaniels** [comment]
Agreed, I'm not sure why the original developer on this project chose that behavior for dump. It could be extremely problematic as you have noted. I will look into this and see if there is something I can do.

**Comment by lmajano** [comment]
FYI, we also have our extension available https://www.ortussolutions.com/products/mongodb-lucee just in case

**Comment by mfr415** [comment]
That extension @lmajano is for caching though right, not querying collections?  I wanted to use that and was willing to pay, but I couldn't get it working.  All the docs around that particular extension suggest using cfcache, etc I'm trying to query mongo to get my data out and insert data into collections, not using it like redis, etc.

**Comment by lmajano** [comment]
This is not our extension. This is the lucee one. Our extension can be found here https://forgebox.io/view/6A03159F-C6DC-4FCC-95A068B1E1A903FA

Let us know in slack box team or community.ortussolutions.com so we can sssist

**Comment by sjdaniels** [comment]
Fixed in [extension version 5.5.1.3](https://download.lucee.org/extension.cfm?groupId=org.lucee&artifactId=mongodb-extension)
